service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java

```diff
@@ -155,4 +158,33 @@
         }
         return result;
     }
+
+    @Override
+    public List<GenericAssayData> getGenericAssayData(String molecularProfileId, String sampleListId,
+                                                    List<String> genericAssayStableIds, String projection)
+        throws MolecularProfileNotFoundException {
+        
+        validateMolecularProfile(molecularProfileId);
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return fetchGenericAssayData(molecularProfileId, sampleIds, genericAssayStableIds, projection);
```

Comment:
I would do as @pvannierop suggests - convert molecularProfileId to list here and call fetchGenericAssayData directly (not getGenericAssayDataInMultipleMolecularProfiles).